app/models/potential_match.rb

```diff
@@ -167,7 +167,15 @@ def self.matching_signup_ids(collection, signup, collection_tag_sets, required_t
                             pluck(:challenge_signup_id).compact
 
       # now add on "any" matches for the required types
-      condition = "any_#{required_types.first.downcase} = 1"
+      condition = {
+        fandom: "any_fandom = 1",
+        character: "any_character = 1",
+        rating: "any_rating = 1",
+        relationship: "any_relationship = 1",
+        category: "any_category = 1",
+        warning: "any_warning = 1",
+        freeform: "any_freeform = 1"
+      }[required_types.first.downcase.to_sym]
```

Comment:
After poking around brakeman some more, I think you can avoid enumerating all tag types in this case by explicitly quoting the column name, like in the following:
```
      column_name = "any_#{required_types.first.underscore}"
      condition = "#{connection.quote_column_name(column_name)} = 1"
```

Reply:
However I think it is more complex and I think the existing code is easier to understand ?

Reply:
The reason I suggested the `quote_column_name` approach in this particular case is because hard-coding all possibilities (whether in a hash or a case statement) makes it more annoying to fix up PRs like #3526, since it adds more places that need to be updated to match the new type name.

(I do prefer the case statement over the hash, but I'd like it better if neither was necessary.)